src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1429 @@
+
+    UT_LOG_INFO("Calling dsHdmiInGetNumberOfInputs");
+    ret = dsHdmiInGetNumberOfInputs(&numInputs);
+    if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: Failed to Get NumberOfInputsPorts:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    }
+       
+    UT_LOG_INFO("Result: NumberOfInputsPorts:[%d]",numInputs);
+
+    if(!numInputs)
+       UT_LOG_INFO("Result: Platform does not supports HdmiIn Ports");
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test get number of input ports.
+*
+* This test function gets HdmiInput ports on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 003@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_get_status(void)
+{
+    gTestID = 3;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    dsHdmiInStatus_t inputstatus;
+    
+    UT_LOG_INFO("Calling dsHdmiInGetStatus");
+    ret = dsHdmiInGetStatus(&inputstatus);
+    if(ret != dsERR_NONE)
+    {
+        UT_LOG_ERROR("Result: Failed to Get HDMI input port status:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    }
+
+    UT_LOG_INFO("Result HDMI input port status isPresented: %s, activeport: %s",
+                 ut_control_GetMapString(bool_mapTable, inputstatus.isPresented),
+                 ut_control_GetMapString(dsHdmiInPort_mapTable, inputstatus.activePort));
+
+    for(int i = 0 ; i < dsHDMI_IN_PORT_MAX ; i++) {
+
+          if(!(inputstatus.isPresented))
+                 continue;
+
+           UT_LOG_INFO("Result: HDMI port connected info isPortConnected: %s, activeport: %s",
+                         ut_control_GetMapString(bool_mapTable, inputstatus.isPortConnected[i]),
+                         ut_control_GetMapString(dsHdmiInPort_mapTable, inputstatus.activePort));
+    }
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to select port.
+*
+* This test function selects HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 004@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_select_port(void)
+{
+    gTestID = 4;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+    
+    dsError_t ret   = dsERR_NONE;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_MAX;
+    int32_t select; 
+    bool audmix;
+    dsVideoPlaneType_t videoplanetype;
+    bool topmost;
+    uint8_t numInputPorts = 0;
+ 
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+
+    printf(" \n================Please Select Inputs==================\n");
+
+    printf("\n*******Enter the port number to select*******\n");
+    readInput(&select);
+
+    if(select <= 0 || select > numInputPorts) 
+    {
+       printf("\nInvalid port selected\n");
```

Comment:
After the failure, the selected variable continues through the test. Maintain the goto exit statement and avoid calling any further functionality. Please Maintain throughout the function

Reply:
UT core team are planning to provide separate function to have menu things. Until then use goto exit or return statements to avoid the invalid choice. We cant pass the invalid choice as such.

Reply:
done